TestSwitchApi/TestSwitchApi/Services/PasswordService.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Security.Cryptography;
+using Microsoft.AspNetCore.Cryptography.KeyDerivation;
+
+namespace TestSwitchApi.Services
+{
+    public class PasswordService : IPasswordService
+    {
+        public string GenerateSalt()
+        {
+            byte[] saltBytes = new byte[128 / 8];
+            using (var rng = RandomNumberGenerator.Create())
+            {
+                rng.GetBytes(saltBytes);
+            }
+
+            string salt = Convert.ToBase64String(saltBytes);
+            return salt;
+        }
+
+        public string HashPassword(string password, string salt)
+        {
+            string hashed = Convert.ToBase64String(KeyDerivation.Pbkdf2(
+                password: password,
+                salt: Convert.FromBase64String(salt),
+                prf: KeyDerivationPrf.HMACSHA256,
+                iterationCount: 5000,
+                numBytesRequested: 256 / 8));
+            return hashed;
+        }
```

Comment:
Just wondering... is there a reason to keep converting everything back to strings?

ie - would it be easier to just keep everything as a byte array always and avoid all the `Convert.ToBase64String` stuff?

Reply:
lol yeah, 

i think it is to do with generating salts in our external salt generator and then inputting them into the DB, it's much easier to do that when you are dealing with strings

Reply:
Have moved the program to use ENV variables to create the admin user